pkg/agent/cniserver/server.go

```diff
@@ -403,8 +415,7 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (*
 	result.IPs = ipamResult.IPs
 	result.Routes = ipamResult.Routes
 	// Ensure interface gateway setting and mapping relations between result.Interfaces and result.IPs
-	gwIPv4 := util.GetIPv4Addr(s.nodeConfig.GatewayConfig.IPs)
-	updateResultIfaceConfig(result, gwIPv4)
+	updateResultIfaceConfig(result, s.nodeConfig.GatewayConfig.IPs)
```

Comment:
I remember we decide to use one IPv4 and one IPv6 field for GatewayConfig, but no more an IP slice?

Reply:
Thanks for the suggestion, we have add PodIPv4CIDR and PodIPv6CIDR for NodeConfig, but haven't changed it for GatewayConfig.
Will update this part by adding separate fields for IPv4 and IPv6.

Reply:
We find that it will be a big change to separate IPv4 and IPv6 fields for both GatewayConfig and InterfaceConfig. The change will also impact current L3 code. We prefer to make a separate patch for this change, what do you think?

Reply:
Sure. Could you add some comment there?

Reply:
Ok. Could we add a TODO comment?

Reply:
Sure, have added a TODO to GatewayConfig definition.  Do you think we need to separate InterfaceConfig IPs? Upstream uses IP slice instead of separate fields for PodIPs (https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L3190) . Maybe we can keep the IP slice in InterfaceConfig ?

Reply:
For interfaceConfig, do we have the cases of multiple v4 IPs of a single interface?

Reply:
I think we do not have such case for now.

Reply:
No strong opinion. Just feel separate v4 and v6 IP fields might make the code simpler. If it is not the case I am fine with a slice.